extensions/wikia/EmbeddableDiscussions/scripts/DiscussionsSettingsModal.js

```diff
@@ -0,0 +1,238 @@
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+
+				$('#embeddable-discussions-settingsmodal-sort-latest').removeClass(
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+			}
+		}
+
+		function openModal() {
+			$.when(
+				loader({
+					type: loader.MULTI,
+					resources: {
+						mustache: 'extensions/wikia/EmbeddableDiscussions/templates/SettingsModal.mustache',
+					}
+				}),
+				nirvana.sendRequest({
+					controller: 'EmbeddableDiscussionsApi',
+					method: 'getCommunityPageSettings',
+					type: 'get',
+					format: 'json'
+				})
+			).then(handleRequestsForModal);
+		}
+
+		/**
+		 * Handle messages, render modal and call createComponent
+		 * One of sub-tasks for getting modal shown
+		 * @param {Object} loaderRes
+		 * @param {Object} nirvanaRes
+		 */
+		function handleRequestsForModal(loaderRes, nirvanaRes) {
+			var categories;
+
+			config = nirvanaRes[0];
+
+			categories = config.allCategories;
+
+			// Pre-check all selected categories
+			for (i = 0; i < categories.length; i++) {
+				if (isSelected(categories[i].id, config.selectedCategories)) {
+					categories[i].checked = true;
+				}
+			}
+
+			modalConfig.vars.content = mustache.render(loaderRes.mustache[0], {
+				heading: $.msg('embeddable-discussions-heading'),
+				description: $.msg('embeddable-discussions-description'),
+				sortBy: $.msg('embeddable-discussions-sort-by'),
+				filterBy: $.msg('embeddable-discussions-filter-by'),
+				filterByAll: $.msg('embeddable-discussions-filter-by-all'),
+				latest: $.msg('embeddable-discussions-show-latest-short'),
+				trending: $.msg('embeddable-discussions-show-trending-short'),
+				cancel: $.msg('embeddable-discussions-cancel-button'),
+				done: $.msg('embeddable-discussions-done-button'),
+				categories: categories,
+				allChecked: (config.selectedCategories.length === 0),
+			});
+
+			require(['wikia.ui.factory'], function (uiFactory) {
+				uiFactory.init(['modal']).then(createComponent);
+			});
+		}
+
+		/**
+		 * Check/unceck the 'All' categories input box
+		 * @param value
+		 */
+		function checkAll(value) {
+			$('#embeddable-discussions-settingsmodal-filter-by-all-checkbox').prop('checked', value);
+		}
+
+		function checkOthers(value) {
+			$('.embeddable-discussions-filter-by-list :checkbox').each(function() {
+				if (this.id !== 'embeddable-discussions-settingsmodal-filter-by-all-checkbox') {
+					$(this).prop('checked', value)
+				}
+			});
+		}
+
+		/**
+		 * Creates modal UI component
+		 * One of sub-tasks for getting modal shown
+		 */
+		function createComponent(uiModal) {
+			/* Create the wrapping JS Object using the modalConfig */
+			uiModal.createComponent(modalConfig, processInstance);
+		}
+
+		/**
+		 * CreateComponent callback that finally shows modal.
+		 * Bind tracking events.
+		 * One of sub-tasks for getting modal shown.
+		 */
+		function processInstance(modalInstance) {
+			var $checkboxes, i;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of the unused 'i' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of the unused '$checkboxes' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)